File: src/main/java/models/Package.java
package models;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "packages")
public class Package {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String name;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "repo_id")
    private Repository repository;

    @OneToMany(mappedBy = "_package", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<File> files;

    public Package() {

    }

    public Package(String name, Repository repository) {
        this.name = name;
        this.repository = repository;
        this.files = new ArrayList<>();
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public Repository getRepository() {
        return repository;
    }

    public List<File> getFiles() {
        return files;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setRepository(Repository repository) {
        this.repository = repository;
    }

    public void setFiles(List<File> files) {
        this.files = files;
    }
}

File: src/main/kotlin/models/Package.kt
package models

import java.util.*
import javax.persistence.*

@Entity
@Table(name = "packages")
class Package {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Int = 0

    var name: String? = null

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "repo_id")
    var repository: Repository? = null

    @OneToMany(mappedBy = "package", cascade = [CascadeType.ALL], orphanRemoval = true)
    var files: MutableList<File>? = null

    constructor() {

    }

    constructor(name: String, repository: Repository) {
        this.name = name
        this.repository = repository
        this.files = ArrayList()
    }

    constructor(id: Int) {
        this.id = id
    }

    companion object {
        fun default(): Package {
            val temp = Package(1)
            temp.name = ""
            temp.repository = Repository.empty()
            return temp
        }
    }
}
